src/sys/mman.rs

```diff
@@ -603,3 +603,64 @@ pub fn shm_unlink<P: ?Sized + NixPath>(name: &P) -> Result<()> {
 
     Errno::result(ret).map(drop)
 }
+
+/// Creates and returns a new, or returns an existing, System V shared memory
+/// segment identifier.
+///
+/// For more information, see [`shmget(2)`].
+///
+/// [`shmget(2)`]: https://man7.org/linux/man-pages/man2/shmget.2.html
+pub fn shmget(key: key_t, size: size_t, shmflg: c_int) -> Result<c_int> {
+    Errno::result(unsafe { libc::shmget(key, size, shmflg) })
+}
+
+/// Attaches the System V shared memory segment identified by `shmid` to the
+/// address space of the calling process.
+///
+/// For more information, see [`shmat(2)`].
+///
+/// # Safety
+///
+/// `shmid` should be a valid shared memory identifier and
+/// `shmaddr` must meet the requirements described in the [`shmat(2)`] man page.
+///
+/// [`shmat(2)`]: https://man7.org/linux/man-pages/man2/shmat.2.html
+pub unsafe fn shmat(
+    shmid: c_int,
+    shmaddr: *const c_void,
+    shmflg: c_int,
+) -> Result<*mut c_void> {
+    Errno::result(libc::shmat(shmid, shmaddr, shmflg))
+}
+
+/// Performs the reverse of [`shmat`], detaching the shared memory segment at
+/// the given address from the address space of the calling process.
+///
+/// For more information, see [`shmdt(2)`].
+///
+/// # Safety
+///
+/// `shmaddr` must meet the requirements described in the [`shmdt(2)`] man page.
+///
+/// [`shmdt(2)`]: https://man7.org/linux/man-pages/man2/shmdt.2.html
+pub unsafe fn shmdt(shmaddr: *const c_void) -> Result<()> {
+    Errno::result(libc::shmdt(shmaddr)).map(drop)
+}
+
+/// Performs control operation specified by `cmd` on the System V shared
+/// memory segment given by `shmid`.
+///
+/// For more information, see [`shmctl(2)`].
+///
+/// # Safety
+///
+/// All arguments should be valid and meet the requirements described in the [`shmctl(2)`] man page.
+///
+/// [`shmctl(2)`]: https://man7.org/linux/man-pages/man2/shmctl.2.html
+pub unsafe fn shmctl(
+    shmid: c_int,
+    cmd: c_int,
+    buf: *mut shmid_ds,
+) -> Result<c_int> {
+    Errno::result(libc::shmctl(shmid, cmd, buf))
```

Comment:
It also seems the meaning of the return value changes depending on the specific command, so it is not correct to treat it as errno unconditionally:

Quoting man 2 shmctl on my system:

> A successful IPC_INFO or SHM_INFO operation returns the index of the highest used entry in the kernel's internal array recording information about all shared  memory  segments.   (This  information  can  be  used  with  repeated  SHM_STAT  or SHM_STAT_ANY  operations to obtain information about all shared memory segments on the system.)  A successful SHM_STAT operation returns the identifier of the shared memory segment whose index was given in shmid.  Other operations return 0 on success.